lib/metrics.ts

```diff
@@ -1,91 +1,89 @@
 
-const env = process.env.NODE_ENV || "staging"; // TODO: "staging" is a non-sense word at Clever
+type jsonData = { [key: string]: any };
+type metricLogger = (title: string, data?: jsonData) => void;
 
-let _last_period_pause_ms = 0;
-
-function logger(source) {
+function logger(source): metricLogger {
   const log = new kv.logger(source);
+  const env = process.env._DEPLOY_ENV;
 
-  return (title, type, value) => {
-    log.infoD(title, { type, value, env, via: "process-metrics" });
-  };
-}
-
-// log_memory_usage logs HeapUsed, HeapTotal, and RSS in the kayvee format
-function log_memory_usage(log) {
-  const mem = process.memoryUsage();
-  log("HeapUsed", "gauge", mem.heapUsed);
-  log("HeapTotal", "gauge", mem.heapTotal);
-  log("RSS", "gauge", mem.rss);
-}
-
-// pause_detector is useful for determining if node isn't processing the event loop. There are
-// two common explanations for these pauses:
-// 1. The event loop is monopolized by one long request
-// 2. Node is garbage collecting
-// This works by sleeping for `sleep_time_ms` and then checking how long it's been since the pause
-// detector was last scheduled. If the difference between actual pause and `sleep_time_ms` is larger
-// than `pause_threshold_ms`, then we can infer that something was monopolizing the event loop.
-function start_pause_detector(log, sleep_time_ms, pause_threshold_ms) {
-  let last_time_ms = Date.now();
-
-  // This function gets called every pause_threshold_ms
-  const pause_fn = function () {
-    const current_time_ms = Date.now();
-    // pause_ms represents the "extra" time the server slept
-    const pause_ms = current_time_ms - last_time_ms - sleep_time_ms;
-
-    // If the pause is long enough, log it immediately so we can associate it with an api request
-    if (pause_ms > pause_threshold_ms) {
-      log("Pause Detected", "counter", pause_ms);
-    }
-
-    // Update the variables for the next call
-    _last_period_pause_ms += pause_ms;
-    last_time_ms = current_time_ms;
+  return (title: string, data?: jsonData) => {
+    log.infoD(title, {
+      env,
+      via: "node-process-metrics",
+      ...data,
+    });
   };
-
-  setInterval(pause_fn, sleep_time_ms);
-}
-
-function log_pauses(log) {
-  log("PauseMetric", "gauge", _last_period_pause_ms);
-  _last_period_pause_ms = 0;
 }
 
 // log_metrics logs node process metrics at the specified frequency. It also logs every time the
 // node event loop stops processing all the events for more than a second.
 module.exports.log_metrics = (
-  source,
-  frequency_ms,
-  pause_threshold_ms = 1000
+  source: string,
+  frequency_ms: number = 30_000
 ) => {
   const log = logger(source);
 
-  setInterval(() => log_memory_usage(log), frequency_ms);
-  start_pause_detector(log, 100, pause_threshold_ms);
-  setInterval(() => log_pauses(log), frequency_ms);
+  log_memory_usage(log, frequency_ms);
+  log_event_loop_utilization(log, frequency_ms);
+  log_event_loop_lag(log, frequency_ms);
```

Comment:
Nit: I would suggest naming these functions to indicate that they don't just log once, they actually start an interval that logs repeatedly (`start_logging_*` e.g.)

Reply:
Fixed in [7a88408](https://github.com/Clever/node-process-metrics/pull/32/commits/7a88408a79f6e2beca67a07f299f69fc90c70bc7)